pkg/broker/broker.go

```diff
@@ -1200,6 +1200,19 @@ func (a AnsibleBroker) Update(instanceUUID uuid.UUID, req *UpdateRequest, async
 
 	req.Parameters = a.validateRequestedUpdateParams(req.Parameters, toPlan, si)
 
+	if fromPlanName == toPlanName && len(req.Parameters) == 0 {
+		log.Warningf("Returning without running the APB. No changes were actually requested")
+
+		a.dao.SetState(si.ID.String(), apb.JobState{
+			Token:   "noop-update",
+			State:   apb.StateSucceeded,
+			Podname: "noop-update",
+			Method:  apb.JobMethodUpdate,
+		})
+
+		return &UpdateResponse{"noop-update"}, nil
```

Comment:
This might seem silly, but this would be constantly overwriting the same job I believe. I wonder if generating a UUID for the token even for the `noop` might make sense?